packages/terminal/src/browser/terminal-quick-open-service.ts

```diff
@@ -0,0 +1,124 @@
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import {
+    QuickOpenModel, QuickOpenItem, QuickOpenHandler,
+    QuickOpenOptions, QuickOpenItemOptions, QuickOpenMode,
+    PrefixQuickOpenService, ApplicationShell
+} from '@theia/core/lib/browser';
+import { WidgetManager } from '@theia/core/lib/browser';
+import { TerminalWidget } from './base/terminal-widget';
+import { TERMINAL_WIDGET_FACTORY_ID } from './terminal-widget-impl';
+
+@injectable()
+export class TerminalQuickOpenService implements QuickOpenModel, QuickOpenHandler {
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    readonly prefix: string = 'term ';
+
+    get description(): string {
+        return 'Show All Opened Terminals';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    get all(): TerminalWidget[] {
+        return this.widgetManager.getWidgets(TERMINAL_WIDGET_FACTORY_ID) as TerminalWidget[];
+    }
+
+    async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        const terminalItems: QuickOpenItem[] = [];
+
+        const widgets: TerminalWidget[] = this.all.sort();
+
+        if (widgets.length === 0) {
+            terminalItems.push(new QuickOpenItem({
+                label: 'List of opened terminals is currently empty',
+                run: () => false
+            }));
+            acceptor(terminalItems);
+            return;
+        }
+
+        for (const widget of widgets) {
+            const item = await this.toItem(widget);
+            terminalItems.push(item);
+        }
+        acceptor(terminalItems);
+        return;
+    }
+
+    /**
+     * Converts the terminal widget to the quick open item.
+     * @param {TerminalWidget} widget - the terminal widget.
+     * @returns Promise of the quick open item.
+     */
+    protected async toItem(widget: TerminalWidget): Promise<QuickOpenItem<QuickOpenItemOptions>> {
+
+        const options: QuickOpenItemOptions = {
+            label: widget.id,
+            iconClass: widget.title.iconClass,
+            description: widget.title.label,
+            tooltip: '',
```

Comment:
Do you think we should include a `tooltip`? Maybe the `title` will be appropriate.